harness/determined/launch/wrap_rank.py

```diff
@@ -47,27 +47,32 @@ def main() -> int:
         "rank",
         metavar="RANK",
         help=(
-            "Can be an integer rank or the name of an "
-            "environment variable containing an integer rank."
+            "Can be an integer rank or a comma-separated list of "
+            "names of environment variables which are tried, in order, "
+            "to determine an integer rank."
         ),
     )
     parser.add_argument(
         "script", nargs=argparse.REMAINDER, metavar="SCRIPT...", help="The worker command."
     )
     args = parser.parse_args()
 
+    rank = None
```

Comment:
This looks like a bug.  It is never allowed to be None.

I mean... it should never _be_ None, but this line only confuses a reader.

Reply:
It needs to be defined before the if/else otherwise they are only visible inside the if/else.

https://www.studytonight.com/python-howtos/scope-of-the-variable-initialised-in-the-if-statement

Reply:
I don't think that is true
```
>>> if True:
...     rank = 0
... 
>>> rank
0
```
python is really weird.

Reply:
or basically, scope in python is very different, maybe because it is interpreted.

Reply:
Hmm.  So must have been misunderstanding what lint was saying.   Or maybe it was more related to the prior version of your code where some path resulting it rank not being set which caused the issue.